### ISSUE TITLE
Do we need a signal server still?

### ISSUE THREAD

**Issue by Parasrah** [WeSyncVideo/board#3]
@t-mullen provided the team with [media-time-sync](https://github.com/t-mullen/media-time-sync), a library to sync media elements taking into account the latency between peers. The purpose of this issue is to discover if we still need to develop a signal server in order to initialise the p2p connection, or if this is made redundant by the library

**Comment by t-mullen** [comment]
The library has sync messages that need to be shared between peers.

WebTorrent handles it's own signalling, but we might want to consider running our own webtorrent-enabled tracker eventually.

There still needs to be some way of exchanging the magnet link.

**Comment by t-mullen** [comment]
Webtorrent manages it's own temporary simple-peer connections, you can't use them for sending your other data without some heavy modification.

We should really try and get an MVP working; fallbacks and optimizations like this can always be added later.

**Comment by Parasrah** [comment]
I agree, I have the fancy features (like chat, fallbacks etc) scheduled for right before a `1.0` release. If we need communication between clients for 
a) your library
b) exchanging the magnet link

Then I would prefer to do this P2P. Personally I'd prefer to do this via
1) A signal server written in a way as to not expose any information about the other clients in the pool unless you have an identifier for one of them
2) A `simple-peer` connection established using the signal server for temporary communication

This way we can limit the amount of information travelling through a server. I really want to prioritise peoples privacy and the security of the peer connections. Does `simple-signal` make those guarantees? I think I remember the reason I didn't use it previously was because it didn't batch messages when the host hasn't connected yet / disconnects briefly? You would know better than I would if that's even necessary though
